pulp_ansible/app/migrations/0047_add_repo_cv_index.py

```diff
@@ -0,0 +1,120 @@
+from django.contrib.postgres import search as psql_search
+from django.db import migrations
+
+
+#  A view that does a view things:
+#   * cartesian product on collectionversions, content and repository_content
+#   * reduce the product via a where clause down to combinations that have matching IDs
+#   * subquery to make a count of signatures attached to the CV and the repo
+#   * subquery to make a count of deprecations related to the CV
+#   * The app code has an unmanaged model that makes this view look like any
+#     other model and is compatible with filtersets and serializers.
+#
+# Why?
+#   * The x-repo search requirements want the same CV listed N times per
+#     repositories containing the CV.
+#   * Many models and objects and concepts desired for cross repo search
+#     filtering are not directly related to repositories.
+#   * Using the queryset language to build the counts for signatures and
+#     deprecations were drifting from wizardry into impossibility.
+#   * Creating a real index model requires too many hooks in various
+#     obscure workflows for CRUD and new devs will always make mistakes.
+
+CREATE_REPOSITORY_COLLECTIONVERSION_VIEW = '''
+CREATE OR REPLACE VIEW repository_collection_version AS
```

Comment:
Could't this be a materialized view?

Using `VIEW` will make all the columns to be computed on every query.
Using `MATERIALIZED VIEW` will create a physical temporary table to store the results of the view and will compute it every time any indexed table is updated, making it faster.

I am not sure if there are any caveats on using it with pulp.

Reply:
When I read the documentation at https://www.postgresql.org/docs/current/rules-materializedviews.html my iterpretation was that the only way to update* data in the materialized view is to call `REFRESH MATERIALIZED VIEW` periodically or anytime a table changes. I was explicitly trying to avoid writing hooks throughout the code that could catch all possible CRUD actions to then update the view.